### PR TITLE
Add support for `--outDir` option

### PR DESCRIPTION
**Pull request by apple-yagi**
I created a Pull Request because I wanted the outputFolder option available in `typed-css-modules` to also be present in `happy-css-modules`. Could you please review it?

### CODE REVIEW

**Comment by mizdra** [comment]
@apple-yagi Question: how does tsc or tsserver load `*.d.ts`? 

For example, imagine the following project.

repo: https://github.com/mizdra/hcm-output-folder-test

```
+ package.json
+ tsconfig.json
+ src
| + Button.tsx
| + Button.module.css
+ dist
  + Button.module.css.d.ts
  + Button.module.css.d.ts.map
```

```json
// package.json
{
  "scripts": {
    "dev:hcm": "hcm -o dist 'src/**/*.module.css'"
  }
}
```

```tsx
// src/Button.tsx
import styles from from './Button.module.css';
export function Button() {
  return <button class={styles.button}>Click me!</button>
}
```

We need to tell tsc and tsserver that the type definition file for `Button.module.css` is `Button.module.css.d.ts`. How do we achieve that?

I believe I need to use `paths` in tsconfig.json and rewrite import speficier to non-relative paths to achieve that. Is that what you expect?

- https://github.com/mizdra/hcm-output-folder-test/commit/b0f2dd114900d6e07fcc9172dafc70add7f7ba6b

To clarify my opinion: I am positive about implementing `--outputFolder`. However, if there is work that users need to do to use `--outputFolder`, it needs to be documented. I want to be clear about it.

**Comment by apple-yagi** [comment]
@mizdra Thank you for checking.

 I will resolve the issue by using the compilerOptions.rootDirs in tsconfig. By doing so, I can use the outputFolder without having to change paths and such.

I've created a Pull Request, so please give it a try.
Thank you!

https://github.com/mizdra/hcm-output-folder-test/pull/1

**Comment by mizdra** [comment]
Wow, this is a good solution. I am going to document the solution. Thank you!

File: packages/happy-css-modules/src/cli.ts [hunk 1]
```diff
@@ -78,6 +78,11 @@ export function parseArgv(argv: string[]): RunnerOptions {
       default: 'info' as RunnerOptions['logLevel'],
       describe: 'What level of logs to report.',
     })
+    .option('outputFolder', {
```
Comment (mizdra) [inline packages/happy-css-modules/src/cli.ts:81]: NITS: happy-css-modules values compatibility with existing tools. For this reason, I would like it to have the same option name (`outDir`) as typed-css-modules.

Can you please rename it?

Comment (apple-yagi) [reply packages/happy-css-modules/src/cli.ts]: I have modified it! https://github.com/mizdra/happy-css-modules/pull/270/commits/65313471395b996962329596add2fbe5c548fc75

Comment (mizdra) [reply packages/happy-css-modules/src/cli.ts]: Thank you!

File: packages/happy-css-modules/src/emitter/dts.ts [hunk 2]
```diff
@@ -1,23 +1,36 @@
 import { EOL } from 'os';
-import { basename, parse, join } from 'path';
+import path, { basename, parse, join } from 'path';
 import camelcase from 'camelcase';
 import { SourceNode, type CodeWithSourceMap } from '../library/source-map/index.js';
 import { type Token } from '../locator/index.js';
 import { type LocalsConvention } from '../runner.js';
 import { getRelativePath, type DtsFormatOptions } from './index.js';
 
+const CURRENT_WORKING_DIRECTORY = process.cwd();
```
Comment (mizdra) [inline packages/happy-css-modules/src/emitter/dts.ts:9]: It is undesirable to fix the current directory to `process.cwd()`. happy-css-modules can explicitly specify the current directory using the `cwd` option of `runner` function. The `runner` function is exposed as a Node.js API.

- https://github.com/mizdra/happy-css-modules?tab=readme-ov-file#nodejs-api-experimental

Therefore, the `outDir` option must also follow the `cwd` option.

Could you modify your code to allow the following test cases to pass? If this is difficult, I can work on your behalf :)

- https://github.com/mizdra/happy-css-modules/pull/270/commits/047c625d64667da5f7050ea7819c9d3714785839

File: README.md [hunk 3]
```diff
@@ -60,6 +60,7 @@ Options:
       --cache                Only generate .d.ts and .d.ts.map for changed files.                                  [boolean] [default: true]
       --cacheStrategy        Strategy for the cache to use for detecting changed files.[choices: "content", "metadata"] [default: "content"]
       --logLevel             What level of logs to report.                            [choices: "debug", "info", "silent"] [default: "info"]
+  -o, --outputFolder         Output folder for generated files.                                                                     [string]
```
Comment (mizdra) [inline README.md:63]: NITS: Please update the name and description of this option.

File: packages/happy-css-modules/src/emitter/dts.ts (hunk 2 above)
Comment (apple-yagi) [reply packages/happy-css-modules/src/emitter/dts.ts:9]: Oh, I see.
I wanted to write a test too but didn't know how to do it, I will modify it to use the cwd option.

Comment (mizdra) [reply packages/happy-css-modules/src/emitter/dts.ts:9]: Thank you.

My advice: [runner.test.ts mocks `console.log` and `console.error`](https://github.com/mizdra/happy-css-modules/blob/047c625d64667da5f7050ea7819c9d3714785839/packages/happy-css-modules/src/runner.test.ts#L26-L29). So if you call `console.log` or `console.error` to debug print, nothing will be output. Use `console.dir` instead.

File: README.md (hunk 3 above)
Comment (apple-yagi) [reply README.md]: https://github.com/mizdra/happy-css-modules/pull/270/commits/fc1fdd64b09b848037f80a1659dbd951d1519e1c

File: packages/happy-css-modules/src/emitter/dts.ts (hunk 2 above)
Comment (apple-yagi) [reply packages/happy-css-modules/src/emitter/dts.ts]: I have tried to fix it and would appreciate a review! https://github.com/mizdra/happy-css-modules/pull/270/commits/d170ef86a46c3ad673099203117f0c86d304ad02

**Review by mizdra — APPROVED** [review]
LGTM. Thank you for your great contribution! Sorry for the late review.

**Comment by mizdra** [comment]
This feature is shipped in v3.2.0.

https://github.com/mizdra/happy-css-modules/releases/tag/v3.2.0

**Pull request merged**